codalib/util.py

```diff
@@ -282,19 +269,19 @@ def updateQueue(destinationRoot, queueDict, debug=False):
     """
 
     attrDict = bagatom.AttrDict(queueDict)
-    url = urlparse.urljoin(destinationRoot, "APP/queue/" + attrDict.ark + "/")
+    url = urllib.parse.urljoin(destinationRoot, "APP/queue/" + attrDict.ark + "/")
     queueXML = bagatom.queueEntryToXML(attrDict)
     urlID = os.path.join(destinationRoot, attrDict.ark)
     uploadXML = bagatom.wrapAtom(queueXML, id=urlID, title=attrDict.ark)
     uploadXMLText = '<?xml version="1.0"?>\n' + etree.tostring(
         uploadXML, pretty_print=True
-    )
+    ).decode()
```

Comment:
Noticing how this PUTs `data` of `str` type versus how you are now sending `bytes` in `sendPREMISEvent`, I was looking at [`urllib.request.Request`](https://docs.python.org/3/library/urllib.request.html#urllib.request.Request) and it seems `str` is not a valid type for `data`. To verify this, I tried POSTing a `str` of XML to our dev Coda instance and got `TypeError: POST data should be bytes, an iterable of bytes, or a file object. It cannot be of type str.` Looks like we need to make sure anywhere we are sending `data` via `Request` that we are using `bytes`.